dealer/dealer.go

```diff
@@ -149,8 +167,16 @@ loop:
 		case <-d.recvLoopStop:
 			break loop
 		default:
-			// no need to hold the reconnectLock since reconnection happens in this routine
+			// no need to hold the connMu since reconnection happens in this routine
```

Comment:
This comment is in contrast with what happens below, is it required to lock for reading?

Reply:
Removed the lock